src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
> If Desktop has the same issue, then these kind of sockets won't work there too now.

That's what I thought too, which means we have a bug in our original implementation (/cc @SidharthNabar @davidsh).

Regarding this PR: since we don't have tests using other AFs, I recommend taking the less risky change which is to add a specific check for AF.Unix and leaving the code paths unaltered for different types of AFs. At least, that way we have only 2 different behaviors between AF.Unix and AF.InterNetwork*.